### PR TITLE
A benchmark suite for the `getBlock` and `getBlocks` RPC calls

### PR DESCRIPTION
**Pull request by steveluscher**
#### Summary of Changes

```shell
     Running `/home/sol/src/agave/target/debug/solana-accounts-cluster-bench --identity /home/sol/.config/solana/id.json --iterations 0 --url l --rpc-bench block --rpc-bench blocks`
[2024-12-06T23:51:01.249719471Z INFO  solana_accounts_cluster_bench] Targeting http://localhost:8899
[2024-12-06T23:51:01.254262179Z INFO  solana_accounts_cluster_bench] Starting balance(s): [666938136880]
[2024-12-06T23:51:01.255387794Z INFO  solana_accounts_cluster_bench] creating 4 new
[2024-12-06T23:51:01.259331039Z INFO  solana_accounts_cluster_bench] txs: 4
[2024-12-06T23:51:01.284633098Z INFO  solana_accounts_cluster_bench] ids: 4
[2024-12-06T23:51:02.293571706Z INFO  solana_accounts_cluster_bench] creating 4 new
[2024-12-06T23:51:02.294493610Z INFO  solana_accounts_cluster_bench] txs: 4
[2024-12-06T23:51:02.317415390Z INFO  solana_accounts_cluster_bench] ids: 4
[2024-12-06T23:51:02.819277816Z INFO  solana_accounts_cluster_bench] creating 4 new
[2024-12-06T23:51:02.820025324Z INFO  solana_accounts_cluster_bench] txs: 4
[2024-12-06T23:51:02.834647911Z INFO  solana_accounts_cluster_bench] ids: 4
[2024-12-06T23:51:03.339562268Z INFO  solana_accounts_cluster_bench] creating 4 new
[2024-12-06T23:51:03.340272594Z INFO  solana_accounts_cluster_bench] txs: 4
[2024-12-06T23:51:03.355491408Z INFO  solana_accounts_cluster_bench] ids: 4
[2024-12-06T23:51:04.360027223Z INFO  solana_accounts_cluster_bench] creating 4 new
[2024-12-06T23:51:04.360836119Z INFO  solana_accounts_cluster_bench] txs: 4
[2024-12-06T23:51:04.375524212Z INFO  solana_accounts_cluster_bench] ids: 4
[2024-12-06T23:51:04.375561944Z INFO  solana_accounts_cluster_bench] total_accounts_created: 20 total_accounts_closed: 0 tx_sent_count: 20 loop_count: 7 balance(s): [666780919280]
[2024-12-06T23:51:04.876743463Z INFO  solana_accounts_cluster_bench] creating 4 new
[2024-12-06T23:51:04.877447477Z INFO  solana_accounts_cluster_bench] txs: 4
[2024-12-06T23:51:04.901852811Z INFO  solana_accounts_cluster_bench] ids: 4
[2024-12-06T23:51:05.256347975Z INFO  solana_accounts_cluster_bench] t(0) rpc(Block) iters: 792 success: 793 errors: 0
[2024-12-06T23:51:05.256388773Z INFO  solana_accounts_cluster_bench]  t(0) rpc(Block average success_time: 5041 us
[2024-12-06T23:51:05.257692240Z INFO  solana_accounts_cluster_bench] t(0) rpc(Blocks) iters: 796 success: 797 errors: 0
[2024-12-06T23:51:05.257735073Z INFO  solana_accounts_cluster_bench]  t(0) rpc(Blocks average success_time: 5021 us
```

Related: #3242.

### CODE REVIEW

File: accounts-cluster-bench/src/main.rs [hunk 1]
```diff
@@ -586,6 +669,7 @@ fn run_accounts_bench(
     loop {
         if latest_blockhash.elapsed().as_millis() > 10_000 {
             blockhash = poll_get_latest_blockhash(&client).expect("blockhash");
+            slot_height_atom.store(poll_slot_height(&client), Ordering::Relaxed);
```
Comment (steveluscher) [inline accounts-cluster-bench/src/main.rs]: This means the slot height will be updated every 10 seconds, along with the blockhash. This is super dumb, but probably good enough.

File: accounts-cluster-bench/src/main.rs [hunk 2]
```diff
@@ -340,6 +367,57 @@ fn run_rpc_bench_loop(
...
+                        stats.errors += 1;
+                        if last_error.elapsed().as_secs() > 2 {
+                            info!("get_block error: {:?}", e);
+                            last_error = Instant::now();
+                        }
+                    }
+                }
+            }
+            RpcBench::Blocks => {
+                static NUM_BLOCKS_TO_FETCH: u64 = 100;
```
Comment (steveluscher) [inline accounts-cluster-bench/src/main.rs]: inb4 ‘magic number.’
giphy-506291741

File: accounts-cluster-bench/src/main.rs [hunk 3]
```diff
@@ -329,6 +355,7 @@ fn run_rpc_bench_loop(
     program_id: &Pubkey,
     max_closed: &AtomicU64,
     max_created: &AtomicU64,
+    slot_height_atom: &AtomicU64,
```
Comment (alessandrod) [inline accounts-cluster-bench/src/main.rs]: nit: here and everywhere else, I'd call this slot_height, then you can rebind the load result on the same name

**Review by alessandrod — DISMISSED** [review]

File: accounts-cluster-bench/src/main.rs (hunk 2 above)
Comment (t-nelson) [reply accounts-cluster-bench/src/main.rs]: should we use the max allowed response size instead? for seniority of magic, ofc

**Comment by steveluscher** [comment]
Used `MAX_GET_CONFIRMED_BLOCKS_RANGE` in place of magic number. Renamed bindings to `slot_height`.

**Review by t-nelson — DISMISSED** [review]

**Comment by steveluscher** [comment]
Updated this to move the `Measure` line right above the actual RPC call.

**Review by alessandrod — DISMISSED** [review]

**Review by t-nelson — DISMISSED** [review]

**Comment by steveluscher** [comment]
Resolved merge conflicts with `master` ergo review required.

**Review by t-nelson — APPROVED** [review]

**Comment by pgarg66** [comment]
Looks like this PR was merged with CI failures.

**Comment by steveluscher** [comment]
@pgarg66 Yeah; sorry about that. See why, here: https://github.com/anza-xyz/agave/pull/3960#issuecomment-2529348325

**Comment by pgarg66** [comment]
> @pgarg66 Yeah; sorry about that. See why, here: https://github.com/anza-xyz/agave/pull/3960#issuecomment-2529348325

Hey, no worries, thanks for pushing the fix!

**Pull request merged**